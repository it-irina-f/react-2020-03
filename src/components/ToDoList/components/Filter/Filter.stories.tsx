import React from "react";
import { action } from "@storybook/addon-actions";

import { Filter } from "./Filter";

export default {
  title: "Filter Component",
};

export const FilterComponent = () => [
  <Filter key="jsx" filterList={action("filter list")} />,
];
