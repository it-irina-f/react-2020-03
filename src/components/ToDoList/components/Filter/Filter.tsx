import React from "react";
import { Button, Input } from "sancho";
import type { TypeAddForm, TypeFilterMode } from "types/todo";

interface FilterProps {
  changeFilter: TypeAddForm;
}

interface FilterState {
  mode: TypeFilterMode[];
}

const modeFilter = [
  { id: "all", title: "Все", isActive: true },
  { id: "active", title: "Активные", isActive: false },
  { id: "done", title: "Завершенные", isActive: false },
];

export class Filter extends React.Component<FilterProps, FilterState> {
  state = {
    mode: modeFilter,
  };

  clickHandler = (id: string) => {
    this.props.changeFilter(id);
    const updBtns = this.state.mode.map((row) => {
      if (row.id === id) {
        return { ...row, isActive: true };
      }
      return { ...row, isActive: false };
    });

    this.setState({
      mode: updBtns,
    });
  };

  render() {
    return (
      <div>
        {this.state.mode.map((row) =>
          row.isActive === true ? (
            <Button key={"btn_" + row.id} intent="primary">
              {row.title}
            </Button>
          ) : (
            <Button
              key={"btn_" + row.id}
              onClick={() => this.clickHandler(row.id)}
            >
              {row.title}
            </Button>
          )
        )}
      </div>
    );
  }
}
