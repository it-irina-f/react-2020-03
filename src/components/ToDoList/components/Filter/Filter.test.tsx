import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

import { Filter } from "./Filter";

const click = jest.fn();

describe("Test Component Filter", () => {
  it("renders Filter: check all", () => {
    expect(
      renderer.create(<Filter changeFilter={click} />).toJSON()
    ).toMatchSnapshot();
  });

  it("should be able to click the button", () => {
    const element = mount(<Filter changeFilter={click} />);
    element.find('div[role="button"]').at(2).simulate("click");
    expect(click).toHaveBeenCalled();
  });
});
