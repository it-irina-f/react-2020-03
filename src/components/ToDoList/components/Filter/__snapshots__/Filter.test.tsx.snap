// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component Filter renders Filter: check all 1`] = `
.emotion-1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  white-space: nowrap;
  -webkit-appearance: none;
  box-sizing: border-box;
  text-align: center;
  -webkit-font-smoothing: antialiased;
  border: none;
  text-rendering: optimizeLegibility;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
  cursor: pointer;
  box-shadow: none;
  font-weight: 500;
  position: relative;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  border-radius: 0.25rem;
  font-size: 1rem;
  padding: 0 1rem;
  height: 40px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #1971c2;
  color: white;
  background-image: linear-gradient(to top,#1971c2,hsl(208.79999999999995,77.2%,60.1%));
  box-shadow: inset 0 0 0 1px hsla(210,10.8%,14.5%,0.2),inset 0 -1px 1px 0 hsla(210,10.8%,14.5%,0.2);
}

.emotion-1:focus {
  outline: 3px auto hsla(208.79999999999995,77.2%,42.9%,0.8);
}

.emotion-1:focus:not([data-focus-visible-added]) {
  outline: none;
}

.emotion-1[aria-expanded="true"] {
  box-shadow: inset 0 0 0 1px hsla(210,10.8%,14.5%,0.5),inset 0 -1px 1px 0 hsla(210,10.8%,14.5%,0.5);
  background-image: linear-gradient(to top,#1864ab,#1971c2);
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 0;
  margin-left: 0;
}

.emotion-3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  white-space: nowrap;
  -webkit-appearance: none;
  box-sizing: border-box;
  text-align: center;
  -webkit-font-smoothing: antialiased;
  border: none;
  text-rendering: optimizeLegibility;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
  cursor: pointer;
  box-shadow: none;
  font-weight: 500;
  position: relative;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  border-radius: 0.25rem;
  font-size: 1rem;
  padding: 0 1rem;
  height: 40px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: white;
  color: #212529;
  background: linear-gradient(to top,#f1f3f5,white);
  box-shadow: inset 0 0 0 1px hsla(210,10.8%,14.5%,0.2),inset 0 -1px 1px 0 hsla(210,10.8%,14.5%,0.1);
}

.emotion-3:focus {
  outline: 3px auto hsla(208.79999999999995,77.2%,42.9%,0.8);
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
}

.emotion-3[aria-expanded="true"] {
  background: none;
  background-color: #f1f3f5;
  box-shadow: inset 0 0 0 1px hsla(210,10.8%,14.5%,0.3),inset 0 -1px 1px 0 hsla(210,10.8%,14.5%,0.15);
}

<div>
  <div
    className="Button Touchable emotion-1"
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseDownCapture={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchMoveCapture={[Function]}
    onTouchStart={[Function]}
    onTouchStartCapture={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      aria-hidden={false}
      className="emotion-0"
    >
      Все
    </span>
  </div>
  <div
    className="Button Touchable emotion-3"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseDownCapture={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchMoveCapture={[Function]}
    onTouchStart={[Function]}
    onTouchStartCapture={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      aria-hidden={false}
      className="emotion-0"
    >
      Активные
    </span>
  </div>
  <div
    className="Button Touchable emotion-3"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseDownCapture={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchMoveCapture={[Function]}
    onTouchStart={[Function]}
    onTouchStartCapture={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      aria-hidden={false}
      className="emotion-0"
    >
      Завершенные
    </span>
  </div>
</div>
`;
